applications/serial_lte_modem/src/slm_at_fota.c

```diff
@@ -375,15 +443,67 @@ void slm_fota_post_process(void)
 
 void slm_finish_modem_fota(int modem_lib_init_ret)
 {
-	if (handle_nrf_modem_lib_init_ret(modem_lib_init_ret)) {
-		char buf[40];
+	char buf[40];
+#if defined(CONFIG_SLM_FULL_FOTA)
+	int err;
+
+	ARG_UNUSED(modem_lib_init_ret);
+
+	if (fota_type == DFU_TARGET_IMAGE_TYPE_FULL_MODEM) {
+		/* Full fota activation differs from delta modem fota. */
+		LOG_INF("Applying full modem firmware update from external flash\n");
+
+		err = nrf_modem_lib_shutdown();
+		if (err != 0) {
+			LOG_ERR("nrf_modem_lib_shutdown() failed: %d\n", err);
+			return;
+		}
+
+		err = nrf_modem_lib_bootloader_init();
+		if (err != 0) {
+			LOG_ERR("nrf_modem_lib_bootloader_init() failed: %d\n", err);
+			return;
+		}
 
-		LOG_INF("Re-initializing the modem due to"
-				" ongoing modem firmware update.");
+		/* Loading data from external flash to modem's flash. */
+		err = fmfu_fdev_load(fmfu_buf, sizeof(fmfu_buf), flash_dev, 0);
+		if (err != 0) {
+			LOG_ERR("fmfu_fdev_load failed: %d\n", err);
+			return;
+		}
+
+		err = nrf_modem_lib_shutdown();
+		if (err != 0) {
+			LOG_ERR("nrf_modem_lib_shutdown() failed: %d\n", err);
+			return;
+		}
+
+		err = nrf_modem_lib_init();
+		if (err != 0) {
+			LOG_ERR("nrf_modem_lib_init() failed: %d\n", err);
+			return;
+		}
+		fota_stage = FOTA_STAGE_COMPLETE;
+		fota_status = FOTA_STATUS_OK;
+		fota_info = 0;
+		LOG_INF("Full modem firmware update succeeded. Will run new firmware");
+
+		/* Extenal flash needs to be erased and internal counters cleared */
+		err = dfu_target_reset();
+		if (err != 0)
+			LOG_ERR("dfu_target_reset() failed: %d\n", err);
+		else
+			LOG_INF("External flash erase succeeded");
+	}
+#endif
+	if (fota_type != DFU_TARGET_IMAGE_TYPE_FULL_MODEM &&
```

Comment:
Should you switch these around so that full modem fota could be in else, which could be inside #ifdef?

Reply:
I would keep it like this, easier to read.